src/chain/chain.rs

```diff
@@ -238,41 +238,50 @@ impl Chain {
     ///
     /// If the event is a `SectionInfo` or `NeighbourInfo`, it also updates the corresponding
     /// containers.
-    pub fn poll(&mut self) -> Result<Option<AccumulatedEvent>, RoutingError> {
-        if self.state.handled_genesis_event
-            && !self.churn_in_progress
-            && !self.relocation_in_progress
-            && !self.state.split_in_progress
-        {
-            if let Some(event) = self.state.churn_event_backlog.pop_back() {
-                trace!(
-                    "{} churn backlog poll {:?}, Others: {:?}",
-                    self,
-                    event,
-                    self.state.churn_event_backlog
-                );
-                return Ok(Some(AccumulatedEvent::new(event)));
-            }
+    pub fn poll_accumulated(&mut self) -> Result<Option<PollAccumulated>, RoutingError> {
```

Comment:
Maybe this can be renamed back to `poll` as there should be no confusion with all the other `poll_*` functions being private now?

Reply:
`poll` is just not that searchable as there are a bunch of thing doing poll, so I'm thinking a bit more explicit name is probably easier to follow, and clearer to read?

Reply:
Ok,